### PR TITLE
Backport bootstrap fixes to branch-2.2

### PR DESCRIPTION
**Pull request by amoskong**

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -1151,6 +1151,7 @@ def calculate_stats_average(self):
             self.stats['results']['stats_total'] = total_stats
 
     def update_test_details(self, snapshot_uploaded=False):
+        return
```
Comment (roydahan) [inline sdcm/tester.py]: Which fix is this related?

File: sdcm/cluster_aws.py [hunk 2]
```diff
@@ -346,9 +346,12 @@ def __init__(self, ec2_ami_id, ec2_subnet_id, ec2_security_group_ids,
 
     def add_nodes(self, count, ec2_user_data='', dc_idx=0, enable_auto_bootstrap=False):
         if not ec2_user_data:
-            ec2_user_data = ('--clustername %s --bootstrap true '
-                             '--totalnodes %s ' % (self.name,
-                                                   count))
+            if self._ec2_user_data:
+                ec2_user_data = self._ec2_user_data
+            else:
+                ec2_user_data = ('--clustername %s --bootstrap true '
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster_aws.py]: You are right, I will fix master first, then update this PR.

File: sdcm/cluster_aws.py [hunk 3]
```diff
@@ -365,7 +365,7 @@ def add_nodes(self, count, ec2_user_data='', dc_idx=0, enable_auto_bootstrap=Fal
                 seeds = ",".join(node_private_ips)
                 if not seeds:
                     seeds = self.nodes[0].private_ip_address
-            ec2_user_data += ' --seeds %s' % seeds
+            ec2_user_data += ' --seeds %s --bootstrap true' % seeds
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster_aws.py]: You are right.

File: sdcm/tester.py (hunk 1 above)
Comment (amoskong) [reply sdcm/tester.py]: Sorry, I pushed a debug patch in this PR, will remove it.

File: sdcm/cluster_gce.py [hunk 4]
```diff
@@ -200,7 +200,7 @@ def _get_persistent_disk_struct(self, name, disk_size, disk_type='pd-ssd', dc_id
                 },
                 "autoDelete": True}
 
-    def add_nodes(self, count, ec2_user_data='', dc_idx=0):
+    def add_nodes(self, count, ec2_user_data='', dc_idx=0, enable_auto_bootstrap=False):
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster_gce.py]: It will effect `--bootstrap` of ec2 user data, and we set node.
And the default node.enable_auto_bootstrap is false, so we only will update enable_auto_bootstrap of the new added nodes of True.

The node.enable_auto_bootstrap will be used in node.config_setup()

File: sdcm/cluster_aws.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster_aws.py]: You are right.

**Comment by roydahan** [comment]
@amoskong ping
